SharpSource/SharpSource/Diagnostics/EnumWithoutDefaultValueAnalyzer.cs

```diff
@@ -30,27 +28,29 @@ public override void Initialize(AnalysisContext context)
     {
         context.EnableConcurrentExecution();
         context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
-        context.RegisterSyntaxNodeAction(AnalyzeSymbol, SyntaxKind.EnumDeclaration);
+        context.RegisterSymbolAction(AnalyzeSymbol, SymbolKind.NamedType);
     }
 
-    private static void AnalyzeSymbol(SyntaxNodeAnalysisContext context)
+    private static void AnalyzeSymbol(SymbolAnalysisContext context)
     {
-        var enumDeclaration = (EnumDeclarationSyntax)context.Node;
-        var enumMembers = enumDeclaration.DescendantNodes().OfType<EnumMemberDeclarationSyntax>();
+        var symbol = (INamedTypeSymbol)context.Symbol;
+        if (symbol.TypeKind != TypeKind.Enum)
+        {
+            return;
+        }
 
-        var membersOfInterest = enumMembers.Where(en => AcceptedNames.Contains(en.Identifier.ValueText)).ToArray();
+        var membersOfInterest = symbol.GetMembers().Where(en => AcceptedNames.Contains(en.Name)).ToArray();
         if (membersOfInterest.Length == 0)
         {
-            Report(enumDeclaration.Identifier, context);
+            Report(symbol, context);
             return;
         }
 
-        var membersWithZeroValue = membersOfInterest.Where(m => context.SemanticModel.GetDeclaredSymbol(m)?.ConstantValue?.Equals(0) == true);
-        if (!membersWithZeroValue.Any())
+        if (!membersOfInterest.Any(m => ( (IFieldSymbol)m ).ConstantValue is 0))
```

Comment:
Is this a safe cast? Wondering if this would cause any issues in the future if enums are expanded to have a different member type. Perhaps something along the lines of this would be safer? That way the analyzer doesn't cause issues.

```suggestion
        if (!membersOfInterest.Any(m => m is IFieldSymbol { ConstantValue: 0 }))
```